sphinx_immaterial/custom_admonitions.py

```diff
@@ -0,0 +1,320 @@
+"""This module inherits from the generic ``admonition`` directive and makes the
+title optional."""
+from pathlib import Path, PurePath
+import re
+from typing import List, Dict, Any, Tuple, Optional, cast, Type, Union
+from docutils import nodes
+from docutils.parsers.rst import directives
+from docutils.parsers.rst.directives import admonitions
+import jinja2
+import pydantic
+from sphinx.application import Sphinx
+import sphinx.addnodes
+import sphinx.ext.todo
+from sphinx.writers.html5 import HTML5Translator
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util.logging import getLogger
+from sphinx.locale import admonitionlabels, _
+from .inline_icons import load_svg_into_builder_env
+
+logger = getLogger(__name__)
+
+
+class CustomAdmonitionConfig(pydantic.BaseModel):
+    """This data class validates the user's configuration value(s) in
+    :confval:`sphinx_immaterial_custom_admonitions`. Each instantiated object
+    corresponds to a generated custom :rst:dir:`admonition` directive tailored with
+    theme specific options.
+    """
+
+    name: str
+    color: Tuple[int, int, int]
+    icon: str
+    title: str
+    override: bool = False
+
+    def __init__(
+        self,
+        name: str,
+        color: Tuple[int, int, int],
+        icon: str,
+        title: str = None,
```

Comment:
`title: Optional[str] = None`

Reply:
It's not actually optional. It needs to have a value after `super().__init__()` is called.

Reply:
I will remove the `None` default ass that is misleading and would be a invalid value.